app/src/main/java/org/ole/planet/myplanet/ui/viewer/ResourceViewerViewModel.kt

```
@@ -5,20 +5,57 @@ import dagger.hilt.android.lifecycle.HiltViewModel
+
+    suspend fun showResourceRatingDialogIfNeverRated(resourceId: String): Boolean {
+        val userId = userRepository.getUserModel()?.id ?: return false
+        if (isRatingPrompted(userId, resourceId)) {
+            return false
+        }
+
+        val hasRated = if (!userId.isNullOrEmpty()) {
+            try {
+                val summary = ratingsRepository.getRatingSummary("resource", resourceId, userId)
+                summary.userRating != null || summary.existingRating != null
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)